app/policies/transaction_policy.rb

```diff
@@ -24,9 +24,10 @@ def resolve
       if user.administrator?
         scope.all
       else
+        # TODO: when we add other hierarchy types, include them here somehow!
```

Comment:
I've been mulling over how we could make this extendable to any type of hierarchy but I've not thought of a way to do this nicely. Ideally the policy scope could be called with additional context but we aren't able to pass this: 

```
transactions = policy_scope(Transaction, hierarchy: :fund).where(fund: @fund)
```

We could create new policies for each hierarchy `FundTransactionPolicy` and `ProgrammeTransactionPolicy` which inheritly knows what parent table to check.. creating new files just to carry state doesn't feel great.

@leeky do you have any ideas for this?